engine/src/org/pentaho/di/cluster/SlaveServer.java

```diff
@@ -241,7 +242,7 @@ public LogChannelInterface getLogChannel() {
     return log;
   }
 
-  public String getXML() {
+  public synchronized String getXML() {
```

Comment:
This makes absolutely no sense. This method creates a new StringBuilder, appends constants to the string, and returns the toString() of it. No actual instance variables referenced. This should absolutely not be synchronized!!!